test/unit/socket-close-test.ts

```diff
@@ -112,19 +112,13 @@ for (const proto of testProtos("tcp", "ipc", "inproc")) {
         }
         this.slow(200)
 
-        const weak = require("weak-napi") as typeof import("weak-napi")
```

Comment:
Do these tests pass after changing to WeakRef? If so, it might worth it to make a separate PR for this specific change in the tests

Reply:
Lol. I was rebasing to find that out! Lots of the stuff in this pull request has been superceded, but I think there are a few useful things. It's worrying that weak-napi broke the build in the first place though. Any idea why?

Reply:
The issue was caused by outdated dependencies. I forked weak-napi, and now I fixed the tests. The tests seem to be working better than before! So, we might not need these changes
 
https://github.com/zeromq/zeromq.js/pull/623

Reply:
But I'd be still interested in you separating it into another pull request where we can test in isolation.

Reply:
> The issue was caused by outdated dependencies in weak-napi. I forked weak-napi, and now I fixed the tests.

How did the outdated dependencies cause it to fail?

The odd thing is that the `weak-napi` repo and its tests seem to work for me. It only was a problem when included as a dependency of `zeromq.js`.

I'm working on separating it out and I think I figured out the big parts. I like the idea of using standard APIs like `WeakRef` or `FinalizationRegistry` rather than pulling in yet another library!